picard/ui/metadatabox.py

```diff
@@ -184,6 +196,31 @@ def __init__(self, parent):
         self.changes_first_action.setCheckable(True)
         self.changes_first_action.setChecked(config.persist["show_changes_first"])
         self.changes_first_action.toggled.connect(self.toggle_changes_first)
+        self.browser_integration = BrowserIntegration()
+		
+    def get_file_lookup(self):
+        """Return a FileLookup object."""
+        return FileLookup(self, config.setting["server_host"],
+                          config.setting["server_port"],
+                          self.browser_integration.port)
+        
+    def open_link(self, item, tag):
+        lookup = self.get_file_lookup()
+        item = item.text()
+        if tag == LOOKUP_TAGS[0]:
```

Comment:
[**@mineo**](https://github.com/mineo) commented on [8 déc. 2015 09:28 UTC+1](https://github.com/musicbrainz/picard/pull/441/files#r46926000 "2015-12-08T08:28:45Z - Replied by Github Comment Enhancer"):
> You can merge this gigantic case statement with LOOKUP_TAGS because Python functions can be passed around as values like to:
> 
> ```
> LOOKUP_TAGS = {
>    "musicbrainz_recordingid": lookup.recordingLookup,
>    [...]
> }
> ```
> 
> That way, all you need to do here would be `LOOKUP_TAGS[tag](item)` (and change https://github.com/musicbrainz/picard/pull/441/files#diff-2c17307a3110cd12224c4e9082becca5R282 to `if tag in LOOKUP_TAGS.keys()`

Thanks ;)